tests/test_manager.py

```diff
@@ -63,30 +63,31 @@ def test_init_app(self):
         :class:`flask.ext.restless.APIManager` object.
 
         """
-        manager = APIManager()
-        manager.init_app(self.flaskapp, session=self.session)
-        manager.create_api(self.Person, app=self.flaskapp)
+        manager = APIManager(session=self.session)
+        manager.create_api(self.Person)
+        manager.init_app(self.flaskapp)
         response = self.app.get('/api/person')
         assert response.status_code == 200
 
     def test_init_app_split_initialization(self):
         manager = APIManager(session=self.session)
+        manager.create_api(self.Person)
         manager.init_app(self.flaskapp)
-        manager.create_api(self.Person, app=self.flaskapp)
         response = self.app.get('/api/person')
         assert response.status_code == 200
 
     def test_init_multiple(self):
-        manager = APIManager(session=self.session)
```

Comment:
Here is the main point. As far as I understand the specs you must be able to init_app your extension on multiple apps. This is required to support the app factories pattern which is crucial, for instance in unit tests.  This is a great concept because it decouples an extension instance from a specific app instance.
What happens in version 0.16 is that, not only you can init_app the same APIManager instance to different apps (as required by the specs), buy you can even specify *different* APIs configurations for different apps using the *same* APIManager instance. As far as I understand this last bit is not required by the specs and breaks the decoupling between blueprints and apps since now you need to provide the app instance when you create a blueprint and not only later, when you register it to an app instance.

To me it seems more intuitive to have two APIManager instances if you want to have to different configurations and to be able to attach each one to as many app instances as you'd like. This is the approach taken by other popular Flask extensions (e.g. Flask-Login, Flask-SQLAlchemy):  __init__ takes all the required configuration settings, init_app only takes an app as a parameter and apply the necessary changes to the app given the settings.
Nobody says that you should follow the same convention, but it looks convenient to me.